### PR TITLE
gromacs: Intel oneapi 2023 fixes

### PR DESCRIPTION
**Pull request by danielahlin**
I made two changes to build gromacs with Intel oneapi 2023

1. specify g++ path (and introduced dependency on gcc)
2. Disable Intel MKL discovery mechanism for 2023 as it breaks for 2023 and also seems no longer needed

### CODE REVIEW

**Comment by marvinbernhardt** [comment]
Sadly, I know next to nothing about oneapi. @danielahlin , since I stopped using Gromacs recently and you seem to be actively working with it, would you be interested in replacing me as a maintainer? I don't know the process for this, and it is likely not my sole decision.

**Comment by danielahlin** [comment]
> Sadly, I know next to nothing about oneapi. @danielahlin , since I stopped using Gromacs recently and you seem to be actively working with it, would you be interested in replacing me as a maintainer? I don't know the process for this, and it is likely not my sole decision.

@marvinbernhardt, I'm happy to help with maintaining the package. However, I want to mention that I'm not an MD subject matter expert nor regular Gromacs user but rather someone who builds and benchmarks Gromacs frequently.

**Comment by marvinbernhardt** [comment]
> > Sadly, I know next to nothing about oneapi. @danielahlin , since I stopped using Gromacs recently and you seem to be actively working with it, would you be interested in replacing me as a maintainer? I don't know the process for this, and it is likely not my sole decision.
> 
> @marvinbernhardt, I'm happy to help with maintaining the package. However, I want to mention that I'm not an MD subject matter expert nor regular Gromacs user but rather someone who builds and benchmarks Gromacs frequently.

I think that is fine. @alalazo how can we change the maintainer? Just make a PR?

**Comment by alalazo** [comment]
@marvinbernhardt Let's just add @danielahlin GitHub user name to the maintainers directive in this PR

**Comment by danielahlin** [comment]
@marvinbernhardt and @alalazo I've added myself as maintainer as discussed.

**Review by junghans — DISMISSED** [review]

File: var/spack/repos/builtin/packages/gromacs/package.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ class Gromacs(CMakePackage):
     url = "https://ftp.gromacs.org/gromacs/gromacs-2022.2.tar.gz"
     list_url = "https://ftp.gromacs.org/gromacs"
     git = "https://gitlab.com/gromacs/gromacs.git"
-    maintainers("junghans", "marvinbernhardt")
+    maintainers("danielahlin", "junghans", "marvinbernhardt")
```
Comment (marvinbernhardt) [inline var/spack/repos/builtin/packages/gromacs/package.py]: suggestion: remove marvinbernhardt

Comment (junghans) [reply var/spack/repos/builtin/packages/gromacs/package.py]: Done added @eirrgang instead.

**Review by junghans — APPROVED** [review]

**Pull request merged**